src/network/adapter.cpp

```diff
@@ -0,0 +1,101 @@
+    dbus::inventory::pathBase + std::string("/system/network/adapter/");
+
+const std::pair<std::string, std::string>
+    NetworkAdapter::pciLookup(const std::string& vendorId,
+                              const std::string& deviceId)
+{
+    static const size_t vidPrefixLen = 0;
+    /* The line of DID has one tabulation to indent */
+    static const size_t didPrefixLen = 1;
+    static const size_t wordLen = 4;
+    static const size_t sepLen = 2;
+    /* The 0x prefix of the VID: `0x____` */
+    static const size_t hexPrefixLen = 2; 
+
+    std::string line;
+    std::string vendorName;
+    std::string modelName;
+    std::ifstream idfile(pciidsPath);
+    if (!idfile.is_open())
+    {
+        log<level::ERR>("failed to open pid.ids file");
+        return std::pair<std::string, std::string>();
+    }
+
+    while (std::getline(idfile, line))
+    {
+        std::string curVIDstr = line.substr(vidPrefixLen, wordLen);
+        if (vendorName.empty() &&
+            (curVIDstr.compare(vendorId.substr(hexPrefixLen)) == 0))
+        {
+            vendorName = line.substr(vidPrefixLen + wordLen + sepLen);
+            if (line.compare(vidPrefixLen + wordLen, sepLen, "  ") != 0)
+            {
+                log<level::ERR>("pid.ids: wrong vendor line format",
+                                entry("VALUE=%s", line.c_str()));
+            }
+        }	
+        else if (modelName.empty() && line.size() > didPrefixLen)
+        {
+            std::string curDIDstr = line.substr(didPrefixLen, wordLen);
+            if (curDIDstr.compare(deviceId.substr(hexPrefixLen)) == 0)
+            {
+                modelName = line.substr(didPrefixLen + wordLen + sepLen);
+                if (line.compare(didPrefixLen + wordLen, sepLen, "  ") != 0)
+                {
+                    log<level::ERR>("pid.ids: wrong device line format",
+                                    entry("VALUE=%s", line.c_str()));
+                }
+            }
+        }
+
+        if (!modelName.empty() && !vendorName.empty())
+        {
+            break;
+        }
+    }
+    return std::pair<std::string, std::string>(vendorName, modelName);
+}
+
+NetworkAdapter::NetworkAdapter(sdbusplus::bus::bus& bus,
+                               const std::string& name,
+                               const std::string& vendor,
+                               const std::string& device,
+                               const std::string& macAddress) :
+    InventoryItemServer(bus, dbusEscape(inventoryPath + name).c_str()),
+    NetworkInterfaceServer(bus, dbusEscape(inventoryPath + name).c_str()),
+    DecoratorAssetServer(bus, dbusEscape(inventoryPath + name).c_str()),
+    OperationalStatusServer(bus, dbusEscape(inventoryPath + name).c_str())
+{
+    auto [vendorName, modelName] = pciLookup(vendor, device);
+    // xyz.openbmc_project.Inventory.Item
+    prettyName(name);
+    present(true);
+    // xyz.openbmc_project.Inventory.NetworkInterface
```

Comment:
I wonder if Phosphor Network Manager (or how is it called) not confused by this new interfaces?

Reply:
The phosphor-dbus-interfaces already have `xyz.openbmc_project.Inventory.Item.NetworkInterface` and the answer to your question: no. The Inventory doesn't relate anyway with `phosphor-network-manager`.

Reply:
Oh, my bad. I'm too wrong.
Looks again on the `phosphor-networkd`, I noted the `xyz.openbmc_project.Inventory.Item.NetworkInterface` used to watch on the mac-address changes(I guess in BMC context).
And that is too strange for me. I still thought the `xyz.openbmc_project.Inventory` purpose is providing an inventory of Host, not of BMC. I was a little bit confused.

Reply:
Yes, I also noticed that xyz.openbmc_project.Inventory.Item.NetworkInterface used to manager BMC network, and that is why I ask.

Reply:
The `xyz.openbmc_project.Inventory.Item.NetworkInterface` in the current service doesn't impact the phoshpor-networkd because the `Sync mac address with phosphor-inventory-manager` was processed only when `phosphor-networkd`:`sync-mac` is turned on.

Totally I very disliked that `xyz.openbmc_project.Inventory.Item` namespace which uses to provide BMC-specific information, I'd strongly like to avoid that feature.